test/Microsoft.TestPlatform.AcceptanceTests/TelemetryTests.cs

```diff
@@ -38,62 +38,60 @@ public void DiscoverTestsShouldPublishMetrics(RunnerInfo runnerInfo)
     {
         SetTestEnvironment(_testEnvironment, runnerInfo);
 
-        DiscoverTests(runnerInfo.RunnerFramework);
+        DiscoverTests(runnerInfo);
     }
 
-    private void RunTests(string runnerFramework)
+    private void RunTests(RunnerInfo runnerInfo)
     {
-        if (runnerFramework.StartsWith("netcoreapp"))
+        if (runnerInfo.IsNetRunner)
         {
             Assert.Inconclusive("Telemetry API is not supported for .NetCore runner");
             return;
         }
 
         var assemblyPaths = GetAssetFullPath("SimpleTestProject2.dll");
 
-        using var tempDir = new TempDirectory();
         var env = new Dictionary<string, string>
         {
-            [LOG_TELEMETRY_PATH] = tempDir.Path,
+            [LOG_TELEMETRY_PATH] = TempDirectory.Path,
             [TELEMETRY_OPTEDIN] = "1",
             [LOG_TELEMETRY] = "1",
         };
 
         InvokeVsTestForExecution(assemblyPaths, GetTestAdapterPath(), FrameworkArgValue, string.Empty, env);
-        ValidateOutput("Execution", tempDir);
+        ValidateOutput("Execution", TempDirectory);
     }
 
-    private void DiscoverTests(string runnerFramework)
+    private void DiscoverTests(RunnerInfo runnerInfo)
     {
-        if (runnerFramework.StartsWith("netcoreapp"))
+        if (runnerInfo.IsNetRunner)
         {
             Assert.Inconclusive("Telemetry API is not supported for .NetCore runner");
             return;
         }
 
         var assemblyPaths = GetAssetFullPath("SimpleTestProject2.dll");
 
-        using var tempDir = new TempDirectory();
         var env = new Dictionary<string, string>
         {
-            [LOG_TELEMETRY_PATH] = tempDir.Path,
+            [LOG_TELEMETRY_PATH] = TempDirectory.Path,
             [TELEMETRY_OPTEDIN] = "1",
             [LOG_TELEMETRY] = "1",
         };
 
         InvokeVsTestForDiscovery(assemblyPaths, GetTestAdapterPath(), string.Empty, FrameworkArgValue, env);
-        ValidateOutput("Discovery", tempDir);
+        ValidateOutput("Discovery", TempDirectory);
     }
 
-    private void ValidateOutput(string command, TempDirectory tempDir)
+    private void ValidateOutput(string command, TempDirectory TempDirectory)
```

Comment:
I don't think you want this rename. Also, do we want to give the possibility to pass a temp dir or shall we use the base property directly within the method?